presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -427,65 +427,70 @@ private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
             throw new TableNotFoundException(tableName);
         }
 
-        Function<HiveColumnHandle, ColumnMetadata> metadataGetter = columnMetadataGetter(table.get(), typeManager);
-        ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
-        for (HiveColumnHandle columnHandle : hiveColumnHandles(table.get())) {
-            columns.add(metadataGetter.apply(columnHandle));
-        }
+        try {
```

Comment:
To keep the diff cleaner, and to avoid an extra level of indentation, I'd prefer adding a method `doGetTableMetadata()` and have the try on the outside.